src/customFileParser/main.ts

```diff
@@ -1,81 +1,61 @@
-import { SectorMetadata, GeometryIndexHandler, UncompressedValues, RenderedGeometryGroups }
-  from './sharedFileParserTypes';
 import CustomFileReader from './CustomFileReader';
-import { renderedGeometries, renderedGeometryToGroup } from './parserParameters';
+import SceneStats from './../SceneStats';
 
-function createSector(
-  sectorMetadata: SectorMetadata,
-  uncompressedValues: UncompressedValues,
-  geometryIndexHandlers: GeometryIndexHandler[]): Sector {
-
-  const counts = countRenderedGeometries(geometryIndexHandlers);
-  const renderedGeometryGroups: RenderedGeometryGroups = {};
-  renderedGeometries.forEach(renderedGeometry => {
-    const count = counts[renderedGeometry];
-    renderedGeometryGroups[renderedGeometry] = new renderedGeometryToGroup[renderedGeometry](count);
-  });
-
-  const groupNames = ['Box', 'Circle', 'ClosedCone', 'ClosedCylinder', 'ClosedEccentricCone', 'ClosedEllipsoidSegment',
-    'ClosedExtrudedRingSegment', 'ClosedGeneralCylinder', 'ClosedSphericalSegment', 'ClosedTorusSegment', 'Ellipsoid',
-    'ExtrudedRing', 'Nut', 'OpenCone', 'OpenCylinder', 'OpenEccentricCone', 'OpenEllipsoidSegment',
-    'OpenExtrudedRingSegment', 'OpenGeneralCylinder', 'OpenSphericalSegment', 'OpenTorusSegment', 'Ring', 'Sphere',
-    'Torus', 'TriangleMesh', 'InstancedMesh'];
-  groupNames.forEach(name => {
-    unpackGeometry(renderedGeometryGroups, geometryIndexHandlers, uncompressedValues, name);
-  });
-
-  const sectorObject = new Sector(sectorMetadata.sectorBBoxMin, sectorMetadata.sectorBBoxMax);
-  Object.keys(renderedGeometryGroups).forEach(renderedGeometryName => {
-    if (renderedGeometryName === 'TriangleMesh') {
-
-    } else if (renderedGeometryName === 'InstancedMesh') {
-
-    } else {
-      sectorObject.primitiveGroups.push(renderedGeometryGroups[renderedGeometryName]);
-    }
-  });
-
-  return sectorObject;
-}
-
-function parseManySectors(fileBuffer: ArrayBuffer): Sector {
+export default function parseCustomFile(fileBuffer: ArrayBuffer) {
   const fileReader = new CustomFileReader(fileBuffer);
-  const sectors: {[name: string]: any} = {};
-
-  let sectorStartLocation = 0;
-  let sectorByteLength = fileReader.readUint32();
-  let sectorMetadata = fileReader.readSectorMetadata(sectorByteLength);
-  const uncompressedValues = fileReader.readUncompressedValues();
-  let geometryIndexHandlers = fileReader.readSectorGeometryIndexHandlers(sectorStartLocation + sectorByteLength);
-  const rootSector = createSector(sectorMetadata, uncompressedValues, geometryIndexHandlers);
-  sectors[sectorMetadata.sectorId] = rootSector;
+  const idToSectorMap: {[name: string]: Sector} = {};
+  const sceneStats: SceneStats = {
+    numInstancedMeshes: 0,
+    numMergedMeshes: 0,
+  };
+  const mergedMeshMap: any = {};
+  const treeIndexNodeIdMap: number[] = [];
+  const colorMap: THREE.Color[] = [];
+
+  let rootSector = undefined;
+  let uncompressedValues = undefined;
+  const sectorPathToPrimitiveData: any = {};
```

Comment:
Could we use types for this? Add them in parseUtils? I don't know if this is the proper way of doing it, but any is "lazy" type scripting :P (I'm also guilty though :D )

Reply:
I don't think it's used widely enough to be turned into a separate named type (it's only used in two files), but I've added more details.